cellengine/resources/fcs_file.py

```diff
@@ -147,7 +147,7 @@ def events(self):
     def events(self, events):
         self._events = events
 
-    def get_events(self, **kwargs):
+    def get_events(self, inplace: bool = False, **kwargs):
         """
         Fetch a DataFrame containing this file's data.
 
```

Comment:
Unrelated: the docstring for `compensatedQ` is incorrect. This method always uses FCS format (so delete the TSV stuff), and currently never applies the file-internal compensation.

You could add a method to FcsFile like `getFileInternalCompensation` that makes an API request to `GET /fcsfiles/:id` with `fields=+spillString` and returns a Compensation instance, then users can apply that as normal. Then the `compensatedQ` parameter can be deleted entirely.

The `spillString` that we return is usually the same as the `$SPILLOVER` or `SPILL` header keyword, except that we correct several kinds of errors that some instrument manufacturers make.

---

Spent 10 mins writing and deleting thoughts about whether or not we need to support `inplace` for `get_events`, but I think it makes sense since it can be used to overwrite the comped events back to uncomped if the user does `compensation.apply(file, inplace=True)` first.

Reply:
"""
                - compensatedQ (bool): If true, applies the compensation
                    specified in compensationId to the exported events.
                    The numerical values will be unchanged, but the
                    file header will contain the compensation as the spill string.
"""

?

Reply:
I thiiiiink that the spillString is returned by default, even without `fields+=spillString`?

Reply:
`GET /experiments/x/fcsfiles/y` includes the spill string, `GET /experiments/x/fcsfiles` does not. I suggested it here as a slight performance improvement since that'd be the only field you need, but definitely don't need to do it.

Reply:
Perfect; I ran into this while implementing this, so it looks like it should work:

```python
class FcsFile:
    #...
    def get_file_internal_compensation(self) -> Compensation:
        file = ce.APIClient().get_fcs_file(self.experiment_id, self._id)
        return Compensation.from_spill_string(file.spill_string)
    #...
```